joomla/joomla-cms
# [5.1] Revert 40037 and clean up code further

Pull Request for Issue # .

### Summary of Changes
This PR reverts https://github.com/joomla/joomla-cms/pull/40037 because some maintainers are not happy with set static property of the class base on parameters from constructor. I also make further clean up to the code to optimize performance a bit :

- Only load plugin language when it is needed (so in the frontend, even someone logged in with special access level or higher, plugin language file won't be loaded)
- Early returns.

### Testing Instructions

- Use Joomla 4.3 nightly build or any Joomla 4.3-dev installation has guided tours installed
- Update your site to the update package generated by this PR
- Try to run a tour, make sure it is still working OK 

### Actual result BEFORE applying this Pull Request
Works


### Expected result AFTER applying this Pull Request
Works with better code and a minor performance improvement for frontend.

### Link to documentations
Please select:
